types/algoliasearch/index.d.ts

```diff
@@ -1337,81 +1339,88 @@ declare namespace algoliasearch {
      * 'none': no exact on single word query
      * 'word': exact set to 1 if the query word is found in the record
      * 'attribute': exact set to 1 if there is an attribute containing a string equals to the query
-     * https://github.com/algolia/algoliasearch-client-js#exactonsinglewordquery
+     * https://www.algolia.com/doc/api-reference/api-parameters/exactOnSingleWordQuery/
      */
-    exactOnSingleWordQuery?: string;
+    exactOnSingleWordQuery?: "attribute"|"none"|"word";
     /**
      * Specify the list of approximation that should be considered as an exact match in the ranking formula
      * default: ['ignorePlurals', 'singleWordSynonym']
      * 'ignorePlurals': alternative words added by the ignorePlurals feature
      * 'singleWordSynonym': single-word synonym (For example "NY" = "NYC")
      * 'multiWordsSynonym': multiple-words synonym
-     * https://github.com/algolia/algoliasearch-client-js#alternativesasexact
+     * https://www.algolia.com/doc/api-reference/api-parameters/alternativesAsExact/
      */
-    alternativesAsExact?: any;
+    alternativesAsExact?: string[];
     /**
      * If set to 1, enables the distinct feature, disabled by default, if the attributeForDistinct index setting is set.
-     * https://github.com/algolia/algoliasearch-client-js#distinct
+     * https://www.algolia.com/doc/api-reference/api-parameters/distinct/
      */
-    distinct?: any;
+    distinct?: number|boolean;
     /**
      * If set to true, the result hits will contain ranking information in the _rankingInfo attribute.
      * default: false
-     * https://github.com/algolia/algoliasearch-client-js#getrankinginfo
+     * https://www.algolia.com/doc/api-reference/api-parameters/getRankingInfo/
      */
     getRankingInfo?: boolean;
     /**
+     * @deprecated Use `numericAttributesForFiltering` instead
      * All numerical attributes are automatically indexed as numerical filters
      * default: ''
-     * https://github.com/algolia/algoliasearch-client-js#numericattributestoindex
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericAttributesForFiltering/
      */
     numericAttributesToIndex?: string[];
+    /**
+     * All numerical attributes are automatically indexed as numerical filters
+     * default: ''
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericAttributesForFiltering/
+     */
+    numericAttributesForFiltering?: string[];
     /**
      * @deprecated please use filters instead
      * A string that contains the comma separated list of numeric filters you want to apply.
-     * https://github.com/algolia/algoliasearch-client-js#numericfilters-deprecated
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericFilters/
      */
     numericFilters?: string[];
     /**
-     * @deprecated
```

Comment:
this @deprecated should stay

Reply:
I was wondering about this one in particular. In another case I can't remember which it was explicitly stated in the docs that an option has been renamed and should no longer be used. Since it wasn't mentioned in this I have removed it.

But if you're having more information on these properties, I'll believe you and apply changes as requested :)

Reply:
It's more of a "discouraged" in this case, but it doesn't have a @ tag in TypeScript